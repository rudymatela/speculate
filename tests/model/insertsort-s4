_ :: [Int]  (holes: [Int])
_ :: Int  (holes: Int)
_ :: Bool  (holes: Bool)
[] :: [Int]
(:) :: Int -> [Int] -> [Int]
(++) :: [Int] -> [Int] -> [Int]
insert :: Int -> [Int] -> [Int]
sort :: [Int] -> [Int]
elem :: Int -> [Int] -> Bool
ordered :: [Int] -> Bool
all :: (Int -> Bool) -> [Int] -> Bool
False :: Bool
True :: Bool
(==) :: Int -> Int -> Bool
(<=) :: Int -> Int -> Bool
(==) :: [Int] -> [Int] -> Bool
(==) :: Bool -> Bool -> Bool

              sort [] == []
       sort (sort xs) == sort xs
             xs ++ [] == xs
          insert x [] == [x]
             [] ++ xs == xs
   sort (insert x xs) == sort (x:xs)
   insert x (sort xs) == sort (x:xs)
             sort [x] == [x]
      sort (xs ++ ys) == sort (ys ++ xs)
           ordered [] == True
    ordered (sort xs) == True
            elem x [] == False
ordered (insert x xs) == ordered xs
     elem x (sort xs) == elem x xs
      (xs == sort xs) == ordered xs
          ordered [x] == True
        all (x <=) [] == True
        all (x ==) [] == True
      (sort xs == []) == (xs == [])

     [] <= xs
sort xs <= xs


