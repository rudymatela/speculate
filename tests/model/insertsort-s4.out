max expr size  =    4
  |- on ineqs  =    3
  |- on conds  =    3
max  #-tests   =  500
min  #-tests   =   25  (to consider p ==> q true)
max  #-vars    =    2  (for inequational and conditional laws)

_ :: Bool
_ :: Int
_ :: [Int]
False :: Bool
True :: Bool
[] :: [Int]
(:) :: Int -> [Int] -> [Int]
(++) :: [Int] -> [Int] -> [Int]
elem :: Int -> [Int] -> Bool
ordered :: [Int] -> Bool
all :: (Int -> Bool) -> [Int] -> Bool
insert :: Int -> [Int] -> [Int]
sort :: [Int] -> [Int]
(==) :: Int -> Int -> Bool
(<=) :: Int -> Int -> Bool
(<) :: Int -> Int -> Bool
(==) :: Bool -> Bool -> Bool
(==) :: [Int] -> [Int] -> Bool

    ordered (sort xs) == True
ordered (insert x xs) == ordered xs
      (xs == sort xs) == ordered xs
     elem x (sort xs) == elem x xs
      (sort xs == []) == (xs == [])
       sort (sort xs) == sort xs
          insert x [] == [x]
      sort (xs ++ ys) == sort (ys ++ xs)
   sort (insert x xs) == insert x (sort xs)
          sort (x:xs) == insert x (sort xs)

    sort xs <= xs
insert x xs <= x:xs


